core/src/main/java/de/medavis/lct/core/patcher/ComponentMetaDataManager.java

```diff
@@ -0,0 +1,121 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package de.medavis.lct.core.patcher;
+
+import de.medavis.lct.core.metadata.ComponentMetaDataLoader;
+import de.medavis.lct.core.metadata.ComponentMetadata;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ComponentMetaDataManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ComponentMetaDataManager.class);
+    private Collection<ComponentMetadata> componentMetaDataList = List.of();
+
+    public static ComponentMetaDataManager create() {
+        return new ComponentMetaDataManager();
+    }
+
+    private void clear() {
+    }
+
+    private void setComponentMetaDataList(@NotNull Collection<ComponentMetadata> list) {
+        clear();
+        this.componentMetaDataList = List.copyOf(list);
+    }
+
+    public void load(@NotNull URI uri) {
+        LOGGER.info("Loading custom rules from '{}'.", uri);
+
+        try {
+            ComponentMetaDataLoader loader = new ComponentMetaDataLoader();
+            if ("file".equals(uri.getScheme())) {
+                setComponentMetaDataList(loader.load(uri.toURL()));
+            } else {
+
+                setComponentMetaDataList(loader.load(uri.toURL()));
+            }
+        } catch (IOException ex) {
+            throw new LicensePatcherException(ex.getMessage(), ex);
+        }
+    }
+
+    public void load(@NotNull Path file) {
+        if (Files.exists(file)) {
+            load(file.toUri());
+        } else {
+            LOGGER.info("No custom rules file '{}' found.", file);
+        }
+    }
+
+    /**
+     * Validate to be mapped license names against an (official SPDX) set of license names.
+     * <p>
+     * If an unsupported SPDX license was found, a log warning will be written.
+     *
+     * @param supportedLicenseNames Set od (SPDX) supported license names
+     * @return Returns true, when unsupported warnings was found
+     */
+    public boolean validateLicenseMappedNames(@NotNull Set<String> supportedLicenseNames) {
+        AtomicBoolean result = new AtomicBoolean(false);
+
+        componentMetaDataList
+                .stream()
+                .filter(cm -> StringUtils.isNotBlank(cm.mappedName()) && !supportedLicenseNames.contains(cm.mappedName()))
+                .forEach(cm -> {
+                    result.set(true);
```

Comment:
This is a weird use of an `AtomicBoolean`. Also, it would be more convenient to have a single log line listing all warnings. Suggestion:

```java
var componentsWithInvalidLicenses = componentMetaDataList.stream()
                                           .filter(cm -> StringUtils.isNotBlank(cm.mappedName()) && !supportedLicenseNames.contains(cm.mappedName()))
                                           .map(ComponentMetadata::mappedName)
                                           .collect(Collectors.toList());
if (!componentsWithInvalidLicenses.isEmpty()) {
    LOGGER.warn("Your component meta data configuration contains unsupported SPDX names: '{}'.", componentsWithInvalidLicenses);
}
return componentsWithInvalidLicenses.isEmpty();
```

Reply:
Fixed